lib/backend/k8s/k8s.go

```diff
@@ -739,9 +740,55 @@ func (c *KubeClient) deleteGlobalConfig(k *model.KVPair) error {
 }
 
 func (c *KubeClient) getHostConfig(k model.HostConfigKey) (*model.KVPair, error) {
+	if k.Name == "IpInIpTunnelAddr" {
+		n, err := c.clientSet.Nodes().Get(k.Hostname, metav1.GetOptions{})
+		if err != nil {
+			return nil, err
+		}
+
+		ip, ipNet, err := net.ParseCIDR(n.Spec.PodCIDR)
+		if err != nil {
+			return nil, err
+		}
+		// We need to get the IP for the podCIDR and increment it to the
+		// first IP in the CIDR.
+		tunIp := ip.To4()
+		tunIp[3]++
+		ipNet.IP = tunIp
+
+		return &model.KVPair{Key: k, Value: tunIp.String()}, nil
+	}
+
 	return nil, errors.ErrorResourceDoesNotExist{Identifier: k}
 }
 
 func (c *KubeClient) listHostConfig(l model.HostConfigListOptions) ([]*model.KVPair, error) {
+	if l.Name == "IpInIpTunnelAddr" {
```

Comment:
IIRC the host config list options could have a blank name - which would mean enumerate over all hosts.  

I think this needs to be a List operation rather than a Get.